legal-api/src/legal_api/services/filings/validations/alteration.py

```diff
@@ -0,0 +1,37 @@
+# Copyright © 2019 Province of British Columbia
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Validation for alteration filing."""
+from http import HTTPStatus  # pylint: disable=wrong-import-order
+from typing import Dict
+from flask_babel import _ as babel  # noqa: N813, I004, I001, I003
+
+from legal_api.errors import Error
+from legal_api.core.filing import Filing  # noqa: I001
+
+from .common_validations import validate_share_structure
+
+
+def validate(alteration_json: Dict):
+    """Validate the alteration filing."""
+    if not alteration_json:
+        return Error(HTTPStatus.BAD_REQUEST, [{'error': babel('A valid alteration is required.')}])
+    msg = []
+
+    err = validate_share_structure(alteration_json, Filing.FilingTypes.ALTERATION.value)
```

Comment:
i don't think an alteration requires a new share structure to be defined.

Reply:
can you please elaborate a bit more? If we do not have an alteration, then we are requesting a valid one. Then we are validating the share structure in an alteration.

Reply:
Here's a valid alteration, https://app.zenhub.com/files/157936592/5c2896b6-37b6-4e64-aee8-d604abfa01b8/download
it still needs to pass even though there is no share structure, it's an optional part of the filing.

Reply:
Return no error if there is no share structure or a valid share structure. Otherwise, returns the error message.

Reply:
open an issue for that

Reply:
This is the issue opened https://github.com/bcgov/entity/issues/6754